### PR TITLE
Feature/308 distance filter

### PR DESCRIPTION
**Pull request by LauraFritz14**
#308 Distance Filter
- select opc
- in surface properties do:
- set home position (not working without)
- check DistanceFilter
- set FilterDistance

### CODE REVIEW

File: src/PRo3D.Viewer/Viewer/Viewer-Utils.fs [hunk 1]
```diff
@@ -701,23 +728,41 @@ module ViewerUtils =
...
-                    yield input.P2
+                let filterDistanceActive : bool = uniform?FilterByDistance
+                let triangleSizeCheck = (alpha && beta && gamma)
+
+                if filterDistanceActive then
+                    let filterRange : float = uniform?FilterDistance
+                    let homePosition : V3d = uniform?HomePosition
+
+                    let inRange = 
+                        (Vec.Distance(homePosition, input.P0.wp.XYZ)) < filterRange &&
```
Comment (haraldsteinlechner) [inline src/PRo3D.Viewer/Viewer/Viewer-Utils.fs:739]: is wp really world position? this is numerically not stable?

**Comment by haraldsteinlechner** [comment]
as hinted above, should filtering operate on view space as opposed to wp which is used here? i think there is a numerical issue. if so i'd propose:
 - transform homePos to view space, rename it to homePosViewSpace
 - compare it with vp instead of wp

**Comment by LauraFritz14** [comment]
Thank you Harri! I updated the branch with your hints

**Comment by haraldsteinlechner** [comment]
Excellent! I'm merging this one. If there is an asscoiated issue please close it.

**Pull request merged**